### PR TITLE
Infer tags if derivedTags is not ready.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/scorecard/models.dart [hunk 1]
```diff
@@ -350,6 +352,53 @@ class ScoreCardData extends Object with FlagMixin {
...
+    } else if (platformTags.length == 1 &&
+        platformTags.single == KnownPlatforms.flutter) {
+      return <String>[
+        'sdk:${SdkTagValue.flutter}',
+      ];
+    } else if (platformTags.length == 1 &&
+        platformTags.single == KnownPlatforms.web) {
+      return <String>[
+        'sdk:${SdkTagValue.dart}',
+        'runtime:${DartSdkRuntimeValue.web}',
```
Comment (jonasfj) [inline app/lib/scorecard/models.dart]: This also implies `sdk:flutter` and `platform:web`.

Comment (isoos) [reply app/lib/scorecard/models.dart]: IIRC it is platformtags = ['web'] if there was use of `dart:html` or `dart:js`, that doesn't imply usability in flutter.

File: app/lib/search/updater.dart [hunk 2]
```diff
@@ -229,6 +229,19 @@ class _PeriodicUpdateTaskSource implements TaskSource {
...
   @override
   Stream<Task> startStreaming() async* {
+    // Scan for potential derived tag updates at startup.
+    // If the package has known platform tags (previous analysis was successful),
+    // but has no sdk: tags, we'll schedule it to be updated.
+    // TODO: remove this code after the current release becomes stable
+    final missingTagTasks = _snapshot.documents.values
+        .where((pd) =>
+            (pd.platforms != null && pd.platforms.isNotEmpty) &&
+            (pd.tags == null || pd.tags.every((t) => !t.startsWith('sdk:'))))
```
Comment (jonasfj) [inline app/lib/search/updater.dart]: Packages that fail analysis will get updated all the time..

Comment (isoos) [reply app/lib/search/updater.dart]: Conditioning on  `(pd.platforms != null && pd.platforms.isNotEmpty)` means we only try this with packages that were not failing the analysis.

**Pull request merged**